shell/platform/android/io/flutter/embedding/FlutterEngine.java

```diff
@@ -0,0 +1,262 @@
+package io.flutter.embedding;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.res.Resources;
+import android.util.Log;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.flutter.embedding.legacy.FlutterPluginRegistry;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.view.FlutterRunArguments;
+
+/**
+ * A single Flutter execution environment.
+ *
+ * A {@code FlutterEngine} can execute in the background, or it can be rendered to the screen by
+ * using the accompanying {@link FlutterRenderer}.  Rendering can be started and stopped, thus
+ * allowing a {@code FlutterEngine} to move from UI interaction to data-only processing and then
+ * back to UI interaction.
+ *
+ * To start running Flutter within this {@code FlutterEngine}, use {@link #runFromBundle(FlutterRunArguments)}.
+ * The {@link #runFromBundle(FlutterRunArguments)} method may not be invoked twice on the same
+ * {@code FlutterEngine}.
+ *
+ * To start rendering Flutter content to the screen, use {@link #getRenderer()} to obtain a
+ * {@link FlutterRenderer} and then attach a {@link FlutterRenderer.RenderSurface}.  Consider using
+ * a {@link FlutterView} as a {@link FlutterRenderer.RenderSurface}.
+ * TODO(mattcarroll): for the above instructions for RenderSurface to be true, we need to refactor
+ *                    the native relationships and also the PlatformViewsController relationship.
+ */
+public class FlutterEngine implements BinaryMessenger {
+  private static final String TAG = "FlutterEngine";
+
+  private final Resources resources;
+  private final FlutterJNI flutterJNI;
+  private final FlutterRenderer renderer;
+  private final FlutterPluginRegistry pluginRegistry;
+  private final Map<String, BinaryMessageHandler> mMessageHandlers;
```

Comment:
Yeah, I'm systematically getting rid of the hungarian notation.  I've been thinking about composing a style doc for the Android embedding.

Reply:
I think the last time we discussed this we figured we should probably follow the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html) as it is enforceable.
The main objection was to not kill all git history (because we need to rename all the m prefixed members), but since you are killing all history anyway maybe nos is a good opportunity.

Reply:
@amirh If those changes are going to happen in this refactor then I'd like to re-open that conversation so I can understand the rationale for the choice.  I mean on the one hand it sounds obvious that a Google team would use the Google Java Style Guide, but also on the other hand I think Google can introduce rules that are overly restrictive and also tied to historical issues that no longer exist.  Do you know the right group of Flutter devs for us to have this conversation again?

Reply:
at least jason, amir, chris, and maurice, but it should be open to anyone else on the team with an interest.

Reply:
I created an empty doc that I will fill with some thoughts in parallel to this refactor and then have it reviewed by everyone:
https://docs.google.com/document/d/162Z7yn0hbttn8NSOp-vd6DFrDqpa4qhbNc2X4O0Ar3E/edit?usp=sharing